xbmc/addons/AddonInfo.cpp

```diff
@@ -151,7 +414,448 @@ CAddonInfo::CAddonInfo(std::string id, TYPE type)
+          {
+            if (elementsAssets->GetText() != nullptr)
+              m_art[value] = elementsAssets->GetText();
+          }
+          else if (value == "clearlogo")
+          {
+            if (elementsAssets->GetText() != nullptr)
+              m_art[value] = elementsAssets->GetText();
+          }
+        }
+      }
+
+      /* Parse addon.xml "<license">...</license>" */
+      element = child->FirstChildElement("license");
+      if (element && element->GetText() != nullptr)
+        m_license = element->GetText();
+
+      /* Parse addon.xml "<source">...</source>" */
+      element = child->FirstChildElement("source");
+      if (element && element->GetText() != nullptr)
+        m_source = element->GetText();
+
+      /* Parse addon.xml "<broken">...</broken>" */
+      element = child->FirstChildElement("broken");
+      if (element && element->GetText() != nullptr)
+        m_broken = element->GetText();
+
+      /* Parse addon.xml "<language">...</language>" */
+      element = child->FirstChildElement("language");
+      if (element && element->GetText() != nullptr)
+        m_language = element->GetText();
+
+      /* Parse addon.xml "<noicon">...</noicon>" */
+      if (m_icon.empty())
+      {
+        element = child->FirstChildElement("noicon");
+        m_icon = (element && strcmp(element->GetText() , "true") == 0) ? "" : "icon.png";
+      }
+
+      /* Parse addon.xml "<nofanart">...</nofanart>" */
+      if (m_art.empty())
+      {
+        element = child->FirstChildElement("nofanart");
+        if (!element || strcmp(element->GetText(), "true") != 0)
+          m_art["fanart"] = URIUtils::AddFileToFolder(addonPath, "fanart.jpg");
+      }
+
+      /* Parse addon.xml "<size">...</size>" */
+      element = child->FirstChildElement("size");
+      if (element && element->GetText() != nullptr)
+        m_packageSize = StringUtils::ToUint64(element->GetText(), 0);
+
+      /* Parse addon.xml "<news lang="..">...</news>" */
+      const char* strChangelog = nullptr;
+      element = child->FirstChildElement("news");
+      while (element)
+      {
+        const char *lang = element->Attribute("lang");
+        if (lang != nullptr && g_langInfo.GetLocale().Matches(lang))
+        {
+          strChangelog = element->GetText();
+          break;
+        }
+        else if (lang == nullptr || strcmp(lang, "en") == 0 || strcmp(lang, "en_GB") == 0)
+        {
+          strChangelog = element->GetText();
+        }
+
+        element = element->NextSiblingElement("news");
+      }
+      m_changelog = strChangelog ? strChangelog : "";
+    }
+    else
+    {
+      TYPE type = TranslateType(point);
+      if (type == ADDON_UNKNOWN || type >= ADDON_MAX)
+      {
+        CLog::Log(LOGERROR, "CAddonInfo: file '%s' doesn't contain a valid add-on type name (%s)", addonPath.c_str(), point.c_str());
+        return false;
+      }
+
+      m_types.push_back(CAddonType(type, this, child));
+    }
+  }
```

Comment:
The existing tests must be complete reworked after cpluff is removed. Everything related on current test parts must be ignored for this.

If you see a error please describe me detailed what it bring.